Add new device: Elecrow 5.0inch esp32 display

bruxy70 opened the pull request:
5-inch 800*480 resolution LCD display with ESP32 S3 microcontroller

Including the minimal configuration.
Plus a more extensive example provided by Elecrow technician for the video I am making (don't want to keep it for myself). 
With the reference to Elecrow wiki with all parameters.

jesserockz reviewed (changes requested):
I just tried this out and it doesn't work. :see_no_evil: 

- The lvgl image configurations are invalid as they do not have a `src`.
- Dont use the `esp32s3box`, use the generic `esp32-s3-devkitc-1`


Example configurations should not have secrets and passwords and should not make any assumptions about data that may or may not be available.

It is fine to have an lvgl example in here, but maybe it could be a bit more useful with buttons etc to show the touch working.

You should not add extra "soft" sensors like uptime, wifi info etc. Only hardware sensors present on a board should be included.

If you want a `1s` update interval for the counter, use the `interval` component.

bruxy70 commented:
Hi. Thanks for the feedback. For lvgl part I did include the official example I got from the manufacturer. But I agree with you that it was too complicated and partially incorrect, so I made own example with one button and one meter.
I was thinking whether to include the lvgl snippet in general, but I think this is realistically much closer to the real use-case that the generic set of ESPHome graphics functions, and lvgl works much better on the display anyway.

sprior commented:
I have been following this merge because I'm using a slight variation of the Elecrow display.  I was just able to get the LVGL version of the example code working with one exception - the 3 power_meter_input_img's are not defined in the example.  I got it working by commenting out the "- image:" sections of each but it would be nice if this part of the example was fixed so we could see what it's supposed to look like.

bruxy70 commented:
@sprior Sorry, I do not follow. I did change the lvgl example based on @jesserockz feedback 2 days ago - it does not contain the 3 images you mention.
There is now one and that one is defined. I also removed the API to HA. And added a button. 
Or if I do not see it, what line number do you refer to?

sprior commented:
@bruxy70 Ok thanks, I was looking at the version on the website, didn't see the fixed version.  I'm working with the MaTouch ESP32-S3 Parallel TFT with Touch 7“ (1024x600) board which so far I've got working with:
`display:
  - platform: rpi_dpi_rgb
    id: main_display
    color_order: RGB
    invert_colors: True
    update_interval: never
    auto_clear_enabled: false # takes too long to clear the display
    dimensions:
      width: 1024
      height: 600
    de_pin: 40
    hsync_pin: 39
    vsync_pin: 41
    pclk_pin: 42
    pclk_frequency: 16MHz
    #
    hsync_front_porch: 40
    hsync_pulse_width: 48
    hsync_back_porch: 28
    vsync_front_porch: 13
    vsync_pulse_width: 3
    vsync_back_porch: 45
    #
    data_pins:
      red:
        - 45        #r1
        - 48        #r2
        - 47        #r3
        - 21        #r4
        - 14        #r5
      green:
        - 5         #g0
        - 6         #g1
        - 7         #g2
        - 15        #g3
        - 16        #g4
        - 4         #g5
      blue:
        - 8         #b1
        - 3         #b2
        - 46        #b3
        - 9         #b4
        - 1         #b5`

And uses different pins for I2C:
`
i2c:
  sda: GPIO17
  scl: GPIO18
  scan: true`

sprior commented:
@bruxy70 I just noticed that in the latest version of your index.md file you still mention esp32s3box  up in the top Drivers section.

bruxy70 commented:
> @bruxy70 I just noticed that in the latest version of your index.md file you still mention esp32s3box up in the top Drivers section.

Thanks. I missed that one. Fixed. The examples are tested and working, so I hope it should now be ready to go?

bruxy70 commented:
The status says change requested.  But I made the changes. What do I have to update to update the status?

jesserockz commented:
> The status says change requested. But I made the changes. What do I have to update to update the status?

Nothing. Someone (if not me) will come back around when time permits =)

jesserockz commented on src/docs/devices/Elecrow-5inch-ESP32-display/index.md:
````diff
@@ -0,0 +1,402 @@
+### data pins
+
+```yaml
+      red:
+        - 45        #r1
+        - 48        #r2
+        - 47        #r3
+        - 21        #r4
+        - 14        #r5
+      green:
+        - 5         #g0
+        - 6         #g1
+        - 7         #g2
+        - 15        #g3
+        - 16        #g4
+        - 4         #g5
+      blue:
+        - 8         #b1
+        - 3         #b2
+        - 46        #b3
+        - 9         #b4
+        - 1         #b5
````
```suggestion
red:
  - 45        #r1
  - 48        #r2
  - 47        #r3
  - 21        #r4
  - 14        #r5
green:
  - 5         #g0
  - 6         #g1
  - 7         #g2
  - 15        #g3
  - 16        #g4
  - 4         #g5
blue:
  - 8         #b1
  - 3         #b2
  - 46        #b3
  - 9         #b4
  - 1         #b5
```

jesserockz commented on src/docs/devices/Elecrow-5inch-ESP32-display/index.md:
```diff
@@ -0,0 +1,402 @@
+esphome:
+  name: display
+  platformio_options:
+    build_flags: "-DBOARD_HAS_PSRAM"
```
Not required as this is set automatically with the `psram` component when using Arduino
```suggestion
```

jesserockz commented on src/docs/devices/Elecrow-5inch-ESP32-display/index.md:
```diff
@@ -0,0 +1,401 @@
+esphome:
+  name: ${device_name}
+  platformio_options:
+    build_flags: "-DBOARD_HAS_PSRAM"
```
```suggestion
```

jesserockz reviewed (approved):
Thanks for making the changes/simplifications.
Compiled and installed this yaml as is and working nicely.

jesserockz commented on src/docs/devices/Elecrow-5inch-ESP32-display/index.md:
```diff
@@ -0,0 +1,400 @@
+---
+title: CrowPanel 5.0" HMI ESP32 display (by Elecrow)
+date-published: 2024-08-17
```
```suggestion
date-published: 2024-09-03
```

jesserockz commented on src/docs/devices/Elecrow-5inch-ESP32-display/index.md:
```diff
@@ -0,0 +1,400 @@
+
+## GPIO Pinout
+
+### i2c (used for touchscreen)
```
```suggestion
### I²C (used for touchscreen)
```

jesserockz commented on src/docs/devices/Elecrow-5inch-ESP32-display/index.md:
```diff
@@ -0,0 +1,400 @@
+| GPIO19 | i2c SDA     |
+| GPIO20 | i2c SCL     |
+
+### backlight
```
```suggestion
### Backlight
```

jesserockz commented on src/docs/devices/Elecrow-5inch-ESP32-display/index.md:
```diff
@@ -0,0 +1,400 @@
+| ------ | ------------- |
+| GPIO2  | backlight   |
+
+### display
```
```suggestion
### Display
```

jesserockz commented on src/docs/devices/Elecrow-5inch-ESP32-display/index.md:
```diff
@@ -0,0 +1,400 @@
+| GPIO41 | vsync_pin   |
+| GPIO0  | pclk_pin    |
+
+### data pins
```
```suggestion
### Display Data pins
```

jesserockz commented on src/docs/devices/Elecrow-5inch-ESP32-display/index.md:
````diff
@@ -0,0 +1,400 @@
+  - 1         #b5
+```
+
+## Simle configuration
````
```suggestion
## Simple Configuration
```

jesserockz commented on src/docs/devices/Elecrow-5inch-ESP32-display/index.md:
````diff
@@ -0,0 +1,400 @@
+            ESP_LOGI("btn", "Button pressed");
+```
+
+## Example (using LGVL graphics library)
````
```suggestion
## Advanced Configuration (using LGVL graphics library)
```

jesserockz reviewed (commented):
Minor capitalisation and spelling
